### ISSUE TITLE
preview samples show good likeness but ckpt file shows bad one ?

### ISSUE THREAD

**Issue by 1blackbar** [ShivamShrirao/diffusers#71]
### Describe the bug

Results from ckpt vs sample images during training is no the same
trained for 5000k iterations, sampels looked very good with perfec likeness on close head shots and mid shots since 3500, but when converted 4500 to ckpt the likeness is pretty bad even on close head shots , kinda looks like 500 steps not 4500 ? Not sure whats going on, maybe saving multiple times like that messes up training .

Also worth to note i did lot of trainings yuesterday with 15 imgs, and had MUCH better resutls with likeness even with 800 steps and 15 imgs than with 4500 15imgs and todays code with saving checkpoint every 500 steps.There must be something in the code that is degrading results, maybe not everything is merged as it should.
Its not overtrained, its undertrained somehow after ckpt conversion.Overtrained would do great likeness but pretty much only training images or artifacted images, this is not it, in case there will be questions about it.

### Reproduction

train for 5k steps with 15 imgs and export to ckpt when sampe imgs are very good, test ckpt in webui - results in pretty bad likeness that looks like much lower steps.

### System Info

colab

**Comment by shadowlocked** [comment]
Can confirm this from a model trained today.

**Comment by 1blackbar** [comment]
yeah we should compare 1500steps trained without saving in middle vs saving every 500 steps till 1500 and test the 1500 ones from both sessions... but there is something going on there

**Comment by shadowlocked** [comment]
> 

Yesterday's repo is what I am using, but it doesn't function like yesterday, because the trained model gets saved into sub-folders in the stated trained directory. One is called 0 and the next something like 4000. For now I am going to try moving the actual files to where they should be. I solved it earlier by changing the paths to the new unexpected sub-folder.

**Comment by ShivamShrirao** [comment]
May be wrong sub folder is being used for ckpt conversion. Can u verify the latest steps are being used ?

**Comment by zakinp** [comment]
i got the opposite results here, lmao. The preview images are horrible but the results in webui are looking good. It seems related to original ckpt model, iterations, learning rate, "--train_text_encoder", and settings in webui

**Comment by shadowlocked** [comment]
> May be wrong sub folder is being used for ckpt conversion. Can u verify the latest steps are being used ?

Correct folder is being used. The backend code is now saving folder-named checkpoints inside the specified save directory. With a max step of 10k and an actual step of 4k, this guarantees at least two sub-folders will be created - _'0'_ and _'4000'_ (or whatever your specified steps were). This breaks the script.

**Comment by ShivamShrirao** [comment]
There is a next cell after training which assigns the WEIGHTS_DIR variable with the latest weights path. Max steps one.

**Comment by shadowlocked** [comment]
> There is a next cell after training which assigns the WEIGHTS_DIR variable with the latest weights path. Max steps one.

Okay. Might be worth including some text to explain what is going on to people at that point, it took me a while to figure out, and I could not understand why the WEIGHTS_DIR box was at the very end.

**Comment by andreae293** [comment]
i too noticed heavy distorsion and artifact the more i trained, is there a way to avoid those artifacts?

**Comment by ShivamShrirao** [comment]
@andreae293 do not train more. It overfits.

**Comment by andreae293** [comment]
but is this overfitting? the face gets totally distorted and does not resemble the original one at all
those artifacts for me starts after 2000 steps
![01974-3460212814-a photo of a bvcfg woman](https://user-images.githubusercontent.com/111545141/198102994-42ba6709-7311-405c-a53b-a6bd6a0f406a.png)
not sure if this is related to this issue

**Comment by 1blackbar** [comment]
Well this topic is old news, im gettin great results now , using saving checkpoints and usually its there at 2400-2800 steps and 20 images, it even works with 3 images and 800 steps but its not that posable (harder to get different head angles if theyre not in face set for learning) the sample previews are game changing for me, nice to see how model learns and when to stop

**Comment by 1blackbar** [comment]
> but is this overfitting? the face gets totally distorted and does not resemble the original one at all those artifacts for me starts after 2000 steps ![01974-3460212814-a photo of a bvcfg woman](https://user-images.githubusercontent.com/111545141/198102994-42ba6709-7311-405c-a53b-a6bd6a0f406a.png) not sure if this is related to this issue

you can improve these far distance full body shots by giving it like 4-5 images of phbotos of the subjects at the same distance from camera, but overall SD is not good at retaining likeness at this distance even with best learned subjects, inpaint the face with full resolution enabled in webui and move on.

**Comment by andreae293** [comment]
has any progress been made on this issue?
i ran a confrontation between this repo and the xavier one
same amount of steps, same regularization images, same training images, same learning rate but i get really different output and a better likeness with xavier repo

**Comment by Maximus-show** [comment]
> был ли достигнут какой-либо прогресс в этой проблеме? я провел конфронтацию между этим репозиторием и xavier на одно и то же количество шагов, те же изображения регуляризации, те же обучающие изображения, та же скорость обучения, но я получаю действительно другой результат и лучшее сходство с xavier repo

Hi. can you share a link to xavier repo ?

**Comment by andreae293** [comment]
> Hi. can you share a link to xavier repo ?

https://github.com/XavierXiao/Dreambooth-Stable-Diffusion

**Comment by Maximus-show** [comment]
> > Привет. можете ли вы поделиться ссылкой на xavier repo?
> 
> https://github.com/XavierXiao/Dreambooth-Stable-Diffusion

Thanks my friend :)
